src/fides/api/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -1381,18 +1382,67 @@ def upload_manual_webhook_data(
     if not privacy_request.status == PrivacyRequestStatus.requires_input:
         raise HTTPException(
             status_code=HTTP_400_BAD_REQUEST,
-            detail=f"Invalid access manual webhook upload request: privacy request '{privacy_request.id}' status = {privacy_request.status.value}.",  # type: ignore
+            detail=f"Invalid manual webhook access upload request: privacy request '{privacy_request.id}' status = {privacy_request.status.value}.",  # type: ignore
         )
 
     try:
-        privacy_request.cache_manual_webhook_input(access_manual_webhook, input_data)
+        privacy_request.cache_manual_webhook_access_input(
+            access_manual_webhook, input_data
+        )
+    except PydanticValidationError as exc:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY, detail=exc.errors()
+        )
+
+    logger.info(
+        "Access input saved for manual webhook '{}' for privacy_request '{}'.",
+        access_manual_webhook,
+        privacy_request,
+    )
+
+
+@router.patch(
+    PRIVACY_REQUEST_MANUAL_WEBHOOK_ERASURE_INPUT,
+    status_code=HTTP_200_OK,
+    dependencies=[Security(verify_oauth_client, scopes=[PRIVACY_REQUEST_UPLOAD_DATA])],
+    response_model=None,
+)
+def upload_manual_webhook_erasure_data(
```

Comment:
i know you specifically mentioned not DRYing this code up, and that's fine, but this is basically identical to the above endpoint besides `cache_manual_webhook_erasure_input` instead of `cache_manual_webhook_access_input` and then one word differences in log messages. IMO it's worth some minimal effort on DRYing this up for readability and so that we can be sure not to have any drift if/when any of this logic needs to be updated.

Reply:
Good point about DRYing this portion up to clearly show these two endpoints are related. I extracted the shared logic into a function so now we can just do this for each endpoint
```
    _handle_manual_webhook_input(
        action="access",
        connection_config=connection_config,
        privacy_request_id=privacy_request_id,
        db=db,
        input_data=input_data,
    )
```

Reply:
awesome, thank you @galvana!